File: lib/test/structuredfield/parseKey.test.ts
import assert from 'node:assert';
import test from 'node:test';
import { parseKey } from '../../src/structuredfield/parse/parseKey.js';

test('parseKey', () => {
	assert.deepStrictEqual(parseKey(`a123_-.*`), { value: `a123_-.*`, src: `` });
	assert.deepStrictEqual(parseKey(`*a123`), { value: `*a123`, src: `` });

	assert.throws(() => parseKey(`&`), /failed to parse "&" as Key/);
});

File: python/src/test/structuredfield/parseKey_test.py
import unittest

from ...structuredfield.parse.ParsedValue import ParsedValue
from ...structuredfield.parse.parseKey import parseKey
from ...structuredfield.SfException import SfException


class TestDecodeSfItem(unittest.TestCase):
    def test_parseKey(self):
        self.assertEqual(parseKey("a123_-.*"), ParsedValue("a123_-.*", ""))
        self.assertEqual(parseKey("*a123"), ParsedValue("*a123", ""))
        self.assertRaises(SfException, parseKey, "&")
